### PR TITLE
Revert "Disallow data.frame without colnames (#1388)"

### PR DESCRIPTION
**Pull request by kbenoit**
Reverts quanteda/quanteda#1390

### CODE REVIEW

**Comment by kbenoit** [comment]
This replaces empty or NA document variables with `V1`, `V2`, etc. This is the same behaviour as `as.data.frame(matrix(1:6, nrow = 2))` for instance. 

It’s written “smartly” so that if the missing docvar is the third document variable, it will name the missing one `V3` instead of `V1`.

File: tests/testthat/test-corpus.R [hunk 1]
```diff
@@ -466,15 +466,54 @@ test_that("corpus handles NA correctly (#1372)", {
...
+                     dvar1 = 1:5,
+                     dvar2 = letters[22:26],
+                     dvar3 = 6:10,
+                     stringsAsFactors = FALSE)
+    
+    # when one column name is NA
+    names(df)[3] <- NA
+    expect_equal(
+        corpus(df) %>% docvars() %>% names(),
+        c("dvar1", "V2", "dvar3")
```
Comment (koheiw) [inline tests/testthat/test-corpus.R]: I am not sure about this expectation. I think this is more natural to make it `"dvar1", "V1", "dvar3"`, ignoring the named ones. If there column original names are `"V1", NA, "V2"`, then `"V1", "V3", "V2"`.

Or should the new column names be followed by column index?

Comment (kbenoit) [reply tests/testthat/test-corpus.R]: Ah, you just found a potential run-time error - if the original names of the data.frame were `c("V2", NA)`. So we should probably issue one more call to `make.names(names(x, unique = TRUE)`. I can add that.

That's why we should always get a second pair of 👀  for a review!

Comment (koheiw) [reply tests/testthat/test-corpus.R]: If `c("V2", NA)` becomes `c("V2", "V1")`, there will be no name collision.

This is the proposed code.
```
l <- !nzchar(names(x)) | is.na(names(x))
name <- setdiff(paste0("V", seq(length(x))), names(x))
names(x)[l] <- head(name, sum(l))
```
However, it seems that `as.data.frame.matrx()` dose nothing with colnames when at least one of them has a name.

```r
> m <- matrix(1:4, nrow = 2)
> m
     [,1] [,2]
[1,]    1    3
[2,]    2    4

> colnames(m)[2] <- "V1"
> as.data.frame(m)
  NA V1
1  1  3
2  2  4
```

**Comment by kbenoit** [comment]
How about this?
```r
df <- data.frame(thetext = LETTERS[1:5],
                     docID = paste0("txt", 1:5),
                     x = 1:5,
                     V3 = letters[22:26],
                     x = 6:10,
                     stringsAsFactors = FALSE)
    names(df)[c(3, 5)] <- c(NA, "")
    crp <- corpus(df, text_field = "thetext", docid_field = "docID")
names(docvars(crp))
## [1] "V1"   "V3"   "V3.1"
```

I don't think we need to make this too user-friendly, because users should not really be feeding data.frames with no column names to `corpus()`.

**Comment by koheiw** [comment]
I revered my change, because this is what we should do. 
```r
> df <- data.frame(Var.1 = 1:3, Var.3 = 1:3, 1:3, c("a", "b", "c"), fix.empty.names = FALSE)
> colnames(df)[4] <- NA
> df
  Var.1 Var.3   NA
1     1     1 1  a
2     2     2 2  b
3     3     3 3  c
> data.frame(df)
  Var.1 Var.3 Var.3.1 NA.
1     1     1       1   a
2     2     2       2   b
3     3     3       3   c
```
This is only to change `x <- as.data.frame(x)` to `x <- data.frame(x)`. data.frame is terribly mysterious.

**Pull request merged**